### PR TITLE
Gbo

### PR DESCRIPTION
**Pull request by leungcalvin**
Refactored ephemeris, added best known CHIME phase center position, and included some modifications for GBO/TONE outrigger. Currently the TONEAntenna objects are hard coded. The fix is entangled with the long term solution to `chimedb_config` and `get_correlator_inputs`, but if we're hard coding the array coordinates I see no problem with hard coding the small number of manageable TONEAntennas while we figure out that problem.

### CODE REVIEW

File: ch_util/tools.py [hunk 1]
```diff
@@ -1178,6 +1190,12 @@ def get_correlator_inputs(lay_time, correlator=None, connect=True):
             correlator = "K7BP16-0004"
         elif correlator.lower() == "chime":
             correlator = "FCC"
+        elif correlator.lower() == "tone":
+            """A hack to return GBO correlator inputs."""
```
Comment (jrs65) [inline ch_util/tools.py:1194]: This should be a comment not a docstring.

**Review by jrs65 — CHANGES_REQUESTED** [review]
Hi @leungcalvin 

This is looking good, and I don't think it needs a huge amount more work before it's ready to merge. Here are some initial comments:
- Please rebase this into a set of logical commits with messages following our dev guidelines (https://github.com/chime-experiment/Pipeline/blob/master/CONTRIBUTING.md#rules). I'll leave you to figure out exactly what to do, but you need at least these changes in separate commits: changing the default location for CHIME; adding support for changing the observatory to the delay/fringestop routines; and adding support for TONE.
- I'm a bit averse to having a dump of all the antennas in directly, just because it dumps in a lot of redundant info. Can you wrap it up into something which programmatically generates them?
- There's a lot of commented out lines (e.g. old params, print statements, etc). Please tidy those up.
- Please fix up the merge conflicts, I think the CI won't run until you have.

**Comment by leungcalvin** [comment]
> Hi @leungcalvin
> 
> This is looking good, and I don't think it needs a huge amount more work before it's ready to merge. Here are some initial comments:
> 
> * Please rebase this into a set of logical commits with messages following our dev guidelines (https://github.com/chime-experiment/Pipeline/blob/master/CONTRIBUTING.md#rules). I'll leave you to figure out exactly what to do, but you need at least these changes in separate commits: changing the default location for CHIME; adding support for changing the observatory to the delay/fringestop routines; and adding support for TONE.
> * I'm a bit averse to having a dump of all the antennas in directly, just because it dumps in a lot of redundant info. Can you wrap it up into something which programmatically generates them?
> * There's a lot of commented out lines (e.g. old params, print statements, etc). Please tidy those up.
> * Please fix up the merge conflicts, I think the CI won't run until you have.

Hi Richard,

I've addressed these comments in a series of logically-chunked commits.

File: ch_util/ephemeris.py [hunk 2]
```diff
@@ -124,19 +124,16 @@
     STELLAR_S,
 )
 
+# Calvin derived the horizontal position of the geometric center of the focal lines and the elevation of the focal line from survey coordinates:
```
Comment (jrs65) [inline ch_util/ephemeris.py:127]: I think you have a doclib entry about this, right? Please put a reference to it in here.

File: ch_util/tools.py [hunk 3]
```diff
@@ -2151,19 +2221,22 @@ def delay(times, feeds, src, wterm=True, prod_map=None, csd=False, static_delays
...
 
     # Get feed positions / c
     feedpos = get_feed_positions(feeds, get_zpos=wterm) / scipy.constants.c
     feed_delays = np.array([f.delay for f in feeds])
+    # print("feed_delays", feed_delays)
+    # print("feed_pos", *feedpos.T[..., np.newaxis])
 
     # Calculate the geometric delay between the feed and the reference position
     delay_ref = -projected_distance(ha, latitude, src_dec, *feedpos.T[..., np.newaxis])
+    # print("delay_ref", delay_ref)
```
Comment (jrs65) [inline ch_util/tools.py:2239]: Please remove all of these commented out print lines.

**Review by jrs65 — CHANGES_REQUESTED** [review]
Hi Calvin, there's two minor changes suggested, and then I think we're done.

**Comment by jrs65** [comment]
Oh also @leungcalvin when you're done, please rebase this such that only the actual relevant commits are left. I think that should be the three `feat(...)` ones and the `test` one.

**Review by jrs65 — DISMISSED** [review]

**Review by jrs65 — APPROVED** [review]

**Pull request merged**